### PR TITLE
fix: consider 'defaultBranch' setting when doing git clone

### PR DESCRIPTION
**Pull request by matemijolovic**
## Changes

This PR fixes the issue where the global config flags `bbUseDevelopmentBranch` and `baseBranches` don't work in a specific case.
It manifests as Renovate's inability to find the `renovate.json` config which is available in one of the `baseBranches` or in a `DEVELOPMENT` branch on Bitbucket, but **not** available on main branch which is mistakenly cloned.

For our usecase, this happens when onboarding a new repository - we treat our main branch as a "production" branch and don't merge `renovate.json` inside instantly - due to this bug we are unable to onboard repositories with a development branch in a standardized way.

## Context

https://github.com/renovatebot/renovate/discussions/26805

This is my first contribution to Renovate so please let me know if anything can be further improved. 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

File: lib/util/git/types.ts [hunk 1]
```diff
@@ -26,6 +27,7 @@ export interface LocalConfig extends StorageConfig {
   additionalBranches: string[];
   currentBranch: string;
   currentBranchSha: LongCommitSha;
+  defaultBranch?: string;
```
Comment (secustor) [inline lib/util/git/types.ts:30]: Why is this needed? 
LocalConfig extends StorageConfig.

Comment (matemijolovic) [reply lib/util/git/types.ts]: it's indeed redundant, I've removed it

**Review by secustor — APPROVED** [review]
